supervise.py

```diff
@@ -267,9 +267,13 @@ def stop(self):
         self._signal("d")
 
     def _signal(self, signal):
-        f = open(self._control,"w+")
-        f.write(signal)
-        f.close()
+        try:
+            f = open(self._control,"w+")
+            f.write(signal)
+        except IOError as e:
+            raise ServiceControlError("Unable to write service control file '{0}'".format(e.filename))
+        else:
```

Comment:
You mean something like this?
```
f = None
try:
    f = open(self._control,"w+")
    f.write(signal)
except IOError as e:
    if f is not None:
      try:
        f.close()
       except IOError:
        pass
    raise ServiceControlError("Unable to write service control file '{0}'".format(e.filename))
```
What about splitting the 2 operations in different try/exc blocks?

```
try:
    f = open(self._control,"w+")
except IOError as e:
    raise ServiceControlError("Unable to open service control file '{0}'".format(e.filename))
try:
    f.write(signal)
except IOError as e:
    raise ServiceControlError("Unable to write service control file '{0}'".format(e.filename))
finally:
    f.close()   
```